reference-stores/next/context/Product.js

```diff
@@ -0,0 +1,31 @@
+import { createContext, useState } from 'react'
+import { getSelectedVariant } from 'utils/getSelectedVariant'
+
+export const ProductContext = createContext({})
+
+export const ProductProvider = ({ children, product }) => {
+  const [selectedOptions, setSelectedOptions] = useState(product.variants[0].content.selectedOptions)
```

Comment:
Similar error as the Nuxt.js version: `TypeError: Cannot read property 'selectedOptions' of null` on /collections/all